pkg/koordlet/statesinformer/impl/states_device_linux_test.go

```diff
@@ -35,6 +35,80 @@ import (
 	mock_metriccache "github.com/koordinator-sh/koordinator/pkg/koordlet/metriccache/mockmetriccache"
 )
 
+func Test_getGPUDeviceConditions(t *testing.T) {
+	tests := []struct {
+		name string
+		gpu  *koordletutil.GPUDeviceInfo
+	}{
+		{
+			name: "healthy GPU device",
+			gpu: &koordletutil.GPUDeviceInfo{
+				UUID:        "test-gpu-uuid",
+				Minor:       0,
+				MemoryTotal: 8000,
+				Status: &koordletutil.DeviceStatus{
+					Healthy: true,
+				},
+			},
+		},
+		{
+			name: "unhealthy GPU device with Xid error",
+			gpu: &koordletutil.GPUDeviceInfo{
+				UUID:        "test-gpu-uuid-2",
+				Minor:       1,
+				MemoryTotal: 8000,
+				Status: &koordletutil.DeviceStatus{
+					Healthy:    false,
+					ErrCode:    "XidCriticalError",
+					ErrMessage: "device is unhealthy due to Xid critical error",
```

Comment:
These should match the actual implementaion, for unhealthy gpus, we only have `DeviceHealthCheckNotSupported` or `Xidx` error code here.